### DISCUSSION TITLE
How to Creating a scalable auto-adjusting size inventory

### DISCUSSION THREAD

**Discussion by ShellWen** [NichtStudioCode/InvUI#41] [Q&A] [answered]
Hi there,

I am facing an issue related to the Inventory interface in our project. Currently, we have an implementation called VirtualInventory that does not include the width and height properties. However, the VirtualInventory does have a `size` property.

My goal is to create a custom Inventory implementation that can automatically adjust its size based on the items it contains. I'm seeking guidance on how to achieve this effectively.

Any suggestions or code examples on how to approach this problem would be greatly appreciated. Thank you in advance for your assistance!

**Comment by NichtStudioCode** [comment, marked as answer]
`VirtualInventory` is backed by a one-dimensional `ItemStack[]`. You can resize it by calling `VirtualInventory#resize`. If this doesn't suit your needs, you can also implement `Inventory` yourself
